Description: Fixed building with ffmpeg 8

 Based on commit 9559eb22c61eb866f900226830858ba5be025b7e from
 upstream by Maarten Baert and proposed pull request
 https://github.com/MaartenBaert/ssr/pull/1075/ by Matthew Mah .
 
Author: Petter Reinholdtsen <pere@debian.org>
Bug-Debian: https://bugs.debian.org/1115053
Forwarded: not-needed
Last-Update: 2025-12-26

--- simplescreenrecorder-0.4.4.orig/src/AV/Output/BaseEncoder.cpp
+++ simplescreenrecorder-0.4.4/src/AV/Output/BaseEncoder.cpp
@@ -176,7 +176,11 @@ void BaseEncoder::Init(const AVCodec* co
 
 void BaseEncoder::Free() {
 	if(m_codec_opened) {
+#if !SSR_USE_AVCODEC_CLOSE_DEPRECATED
 		avcodec_close(m_codec_context);
+#else
+		avcodec_free_context(&m_codec_context);
+#endif
 		m_codec_opened = false;
 	}
 }
--- simplescreenrecorder-0.4.4.orig/src/Global.h
+++ simplescreenrecorder-0.4.4/src/Global.h
@@ -257,6 +257,10 @@ inline void atomic_thread_fence_replacem
 // avformat_free_context: lavf 52.96.0 / 52.96.0
 #define SSR_USE_AVFORMAT_FREE_CONTEXT              TEST_AV_VERSION(LIBAVFORMAT, 52, 96, 52, 96)
 
+// avcodec_close deprecated: lavc 60.39.100 / ???
+// - ffmpeg: missing, commit: https://github.com/FFmpeg/FFmpeg/commit/1cc24d749569a42510399a29b034f7a77bdec34e
+#define SSR_USE_AVCODEC_CLOSE_DEPRECATED           TEST_AV_VERSION(LIBAVCODEC, 60, 39, 999, 999)
+
 // av_codec_iterate: lavc 58.10.100 / ???
 #define SSR_USE_AV_CODEC_ITERATE                   TEST_AV_VERSION(LIBAVCODEC, 58, 10, 999, 999)
 // av_lockmgr_register deprecated: lavc 58.9.100 / ???
